# joint-wise values for nan-entry interpolation
joint_repl_values:
  joint7:
    zero:
      cmd: 0.0
      angle: {'min': -0.050417547, 'max': 0.052598482}
  joint8:
    tcp: 'joint_r_laser'
    zero:
      cmd: 0.0
      angle: {'min': -0.0314886537, 'max': 0.033270853}
  jointT0:
    zero:
      cmd: 0.0
      angle: {'min': -0.0653188945, 'max': 0.0614732712}
    moved:
      cmd: 2.0
      angle: {'min': 0.577, 'max': 0.637}
  jointT1:
    zero:
      cmd: -3.142
      angle: {'min': 0, 'max': 0.0382605643}
  jointT2:
    zero:
      cmd: -3.142
      angle: {'min': 0, 'max': 0.0382605643}
  jointT3:
    tcp: 'joint_Tbumper'
    zero:
      cmd: -3.142
      angle: {'min': 0, 'max': 0.0291258907}
  jointI1:
    zero:
      cmd: -3.142
      angle: {'min': 0, 'max': 0.0382605643}
  jointI2:
    zero:
      cmd: -3.142
      angle: {'min': 0, 'max': 0.0652149185}
  jointI3:
    tcp: 'joint_Ibumper'
    zero:
      cmd: -3.142
      angle: {'min': 0, 'max': 0.0708832264}
  jointM1:
    zero:
      cmd: -3.142
      angle: {'min': 0, 'max': 0.0799608775}
  jointM2:
    zero:
      cmd: -3.142
      angle: {'min': 0, 'max': 0.091501851}
  jointM3:
    tcp: 'joint_Mbumper'
    zero:
      cmd: -3.142
      angle: {'min': 0, 'max': 0.0838347478}
  jointR1:
    zero:
      cmd: -3.142
      angle: {'min': 0, 'max': 0.0454248685}
  jointR2:
    zero:
      cmd: -3.142
      angle: {'min': 0, 'max': 0.0482509761}
  jointR3:
    tcp: 'joint_Rbumper'
    zero:
      cmd: -3.142
      angle: {'min': 0, 'max': 0.0973900944}
  jointL1:
    zero:
      cmd: -3.142
      angle: {'min': 0, 'max': 0.0200728147}
  jointL2:
    zero:
      cmd: -3.142
      angle: {'min': 0, 'max': 0.0139034957}
  jointL3:
    tcp: 'joint_Lbumper'
    zero:
      cmd: -3.142
      angle: {'min': 0, 'max': 0.0280540352}

# descriptions for actuator commands
description_mapping:
  joint7: {name: 'wrist abduction', len: 9, offset: 0}
  joint8: {name: 'wrist flexion', len: 9, offset: 12}
  jointT0: {name: 'thumb abduction', len: 12, offset: 22}
  jointT1: {name: 'thumb flexion', len: 36, offset: 35}
  jointT2: {name: 'thumb flexion', len: 36, offset: 35}
  jointT3: {name: 'thumb flexion', len: 36, offset: 35}
  jointI1: {name: 'jointI1 flexion', len: 36, offset: 72}
  jointI2: {name: 'jointI1 flexion', len: 36, offset: 72}
  jointI3: {name: 'jointI1 flexion', len: 36, offset: 72}
  jointM1: {name: 'jointM1 flexion', len: 36, offset: 109}
  jointM2: {name: 'jointM1 flexion', len: 36, offset: 109}
  jointM3: {name: 'jointM1 flexion', len: 36, offset: 109}
  jointR1: {name: 'jointL1R1 flexion', len: 36, offset: 146}
  jointR2: {name: 'jointL1R1 flexion', len: 36, offset: 146}
  jointR3: {name: 'jointL1R1 flexion', len: 36, offset: 146}
  jointL1: {name: 'jointL1R1 flexion', len: 36, offset: 146}
  jointL2: {name: 'jointL1R1 flexion', len: 36, offset: 146}
  jointL3: {name: 'jointL1R1 flexion', len: 36, offset: 146}
  misc: ['home', 'start', 'new sequence', 'reset angle']